companion/lib/Data/StoreBase.ts

```diff
@@ -377,67 +296,134 @@ export class DataStoreBase {
 	}
 
 	/**
-	 * Save/update a key/value pair to the database
+	 * Save/update a key/value pair to the default table
+	 * @param key - the key to save under
+	 * @param value - the object to save
+	 */
+	public setKey(key: string, value: any): void {
+		this.setTableKey(this.defaultTable, key, value)
+	}
+
+	/**
+	 * Save/update a key/value pair to a table
+	 * @param table - the table to save in
 	 * @param key - the key to save under
 	 * @param value - the object to save
-	 * @access public
-	 */
-	setKey(key: number | string | string[], value: any): void {
-		this.logger.silly(`${this.name}_set(${key}, ${value})`)
-
-		if (key !== undefined) {
-			if (Array.isArray(key)) {
-				if (key.length > 0) {
-					const keyStr = key.join(':')
-					const lastK = key.pop()
-
-					// Find or create the parent object
-					let dbObj = this.store
-					for (const k of key) {
-						if (!dbObj || typeof dbObj !== 'object') throw new Error(`Unable to set db path: ${keyStr}`)
-						if (!dbObj[k]) dbObj[k] = {}
-						dbObj = dbObj[k]
+	 */
+	public setTableKey(table: string, key: string, value: any): void {
+		if (table.length > 0 && key.length > 0 && value) {
+			if (typeof value === 'object') {
+				value = JSON.stringify(value)
+			}
+
+			const query = this.store?.prepare(
```

Comment:
these prepared statements could be cached and reused, which might be good for performance. It also might be a bit premature to be thinking about that, so feel free to ignore

Reply:
Yeah I could.  I want to try subbing the hard table with an `@table` param (but a different keyword) and if that works that makes caching these easier.

Reply:
They get cached now.  Couldn't make table a var.